indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -565,6 +565,16 @@ def _process(self):
         return jsonify_data(html=_render_registration_details(self.registration))
 
 
+class RHRegistrationReset(RHManageRegistrationBase):
+    """Reset a registration back to a non-approved status"""
+
+    def _process(self):
```

Comment:
Shouldn't we also check if moderation is enabled? Or are we already ignoring an `approved=False` state update in that case?

Reply:
I don't think it's necessary to check for moderation on the RH, unless you want to raise an explicit error. `update_state` will not do anything if moderation is not enabled:

```python
elif self.state == RegistrationState.complete:
    if approved is False and payment_required is False and moderation_required:
        self.state = RegistrationState.pending
```

Reply:
Fine for me then since we hide the button anyway if moderation is not enabled.